MonoGame.Framework/Graphics/GraphicsDevice.cs

```diff
@@ -413,6 +421,9 @@ protected virtual void Dispose(bool disposing)
                     // Dispose of all remaining graphics resources before disposing of the graphics device
                     GraphicsResource.DisposeAll();
 
+                    // Clear the effect cache.
+                    EffectCache.Clear();

```

Comment:
Cool...  so the only issue remaining now is that `GraphicsResource.DisposeAll()` will dispose resources owned by a second `GraphicsDevice`.  Same drill as all these other changes... the static collection holding all the `GraphicsResource`s really needs to be owned by the `GraphicsDevice` instance.